cli/config/configfile/file_test.go

```diff
@@ -481,6 +481,100 @@ func TestLoadFromReaderWithUsernamePassword(t *testing.T) {
 	}
 }
 
+const envTestUserPassConfig = `{
+	"auths": {
+		"env.example.test": {
+			"username": "env_user",
+			"password": "env_pass",
+			"serveraddress": "env.example.test"
+		}
+	}
+}`
+
+const envTestAuthConfig = `{
+	"auths": {
+		"env.example.test": {
+			"auth": "ZW52X3VzZXI6ZW52X3Bhc3M="
+		}
+	}
```

Comment:
See my other comment about "empty" values here; perhaps we should have that as a test-case to verify that behavior;

```suggestion
	"auths": {
		"env.example.test": {
			"auth": "ZW52X3VzZXI6ZW52X3Bhc3M="
		},
		"https://index.docker.io/v1/": {},
		"https://index.docker.io/v1/access-token": {},
		"https://index.docker.io/v1/refresh-token": {}
	}
```